src/precompile_utils.c

```diff
@@ -187,7 +187,7 @@ static int precompile_enq_specialization_(jl_method_instance_t *mi, void *closur
             if (inferred &&
                 inferred != jl_nothing &&
                 jl_ir_flag_inferred((jl_array_t*)inferred) &&
-                (jl_ir_inlining_cost((jl_array_t*)inferred) == UINT16_MAX)) {
```

Comment:
@vchuravy Could you please elaborate why we have this check and want to precompile sources that aren't inlineable?

Reply:
cc: @vtjnash and @timholy 

My expectation would be that we want a warm cache for inference even for sources we inlined.

Reply:
@timholy or @vtjnash, could you please provide some insights on why we have this check and whether it's still effective nowadays? The equivalent check is implemented in this commit (3a20e6a), but if we can remove the logic, it would simplify this PR very much.

Reply:
Okay, so it's pretty effective, since we hit this condition when the `isa_compileable_sig(linfo.specTypes, linfo.sparam_vals, def)` case in this line holds 
https://github.com/JuliaLang/julia/blob/93df7e28c12a171b0a579f5e7517429def7c1d3b/base/compiler/typeinfer.jl#L344
, which happens pretty frequently.